### ISSUE TITLE
Frame errors in 1D stellar matom model

### ISSUE THREAD

**Issue by jhmatthews** [sirocco-rt/sirocco#805]
When I run the attached 1D stellar model with the dev version, I get errors of the following type
```
Error: check_frame: move_phot :Photon (2) of type (1) not in frame 1
```
This doesn't seem to happen in the non-macro atom version of the same model - which is a little odd. @saultyevil mentioned similar things in a 1D imported model, I think. The errors happen during the ionization cycles, in the photon transport stage. 

[star_macro.pf.txt](https://github.com/agnwinds/python/files/5797392/star_macro.pf.txt)

**Comment by saultyevil** [comment]
Yeah, this is the same error I get for an imported spherical model. I get the same error during the ionization cycles, but I haven't let the model run all the way to spectral cycles yet to see if it happens there.

It doesn't happen for every photon either.

```
Error: check_frame: move_phot :Photon (701) of type (1) not in frame 1
```

This is the first time it happens out of 1e7 photons.

**Comment by saultyevil** [comment]
The error happens when calculating the gradient of the velocity field and only happens for spherical models, since we consider it a special case.

You can see it here

```c
if (zdom[ndom].coord_type == SPHERICAL)
{
  struct photon pnew;
  double v1[3], v2[3], dv[3], diff[3];
  double ds;
  /* choose a small distance which is dependent on the cell size */
  vsub (pp.x, wmain[pp.grid].x, diff);
  ds = 0.001 * length (diff);
  /* calculate the velocity at the position of the photon */
  /* note we use model velocity, which could potentially be slow,
     but avoids interpolating (see #118) */
  model_velocity (ndom, pp.x, v1);

  /* copy the photon and move it by ds, and evaluate the velocity
     at the new point */
  stuff_phot (&pp, &pnew);
  move_phot (&pnew, ds);  // <----- problem is HERE. Replace with vadd call instead?
  model_velocity (ndom, pnew.x, v2);

  /* calculate the relevant gradient */
  if (rel_mode == REL_MODE_FULL)
  {
    observer_to_local_frame_velocity (v2, v1, dv);
    dvds = length (dv) / ds;
  }
  else
  {
    dvds = fabs (dot (v1, pp.lmn) - dot (v2, pp.lmn)) / ds;
  }
}
```

What is happening is that pp and pnew are both in F_LOCAL, i.e. the CMF. So when it's moved that distance ds, move_phot complains as we only want to use move_phot to move photons in the observer frame. James has made a note above the code saying that a on-the-fly method was adopted for spherical coordinates (issue #118), suggesting we figure out how to use the gradient tensor in the future.

But actually, I think we could figure this problem out by avoiding move_phot and using vadd to add the vector ds to the photon position instead, and we could probably upgrade to using a symmetric finite difference approximation to calculate the derivative like in issue #782.

**Comment by kslong** [comment]
Even simpler would be just to set the flag to the correct value.  That is what is done elsewhere in the code.

**Comment by saultyevil** [comment]
Yep, that sounds good. I've added this, but the internet is mysteriously down in Southampton so I'll create a pull request when everything is back!

**Comment by saultyevil** [comment]
This has been fixed with PR #807.
